meson_options.txt

```diff
@@ -0,0 +1,98 @@
+# Nix project build options
+#============================================================================
+
+# dirs
+#============================================================================
+
+option(
+    'nixstoredir',
+    type : 'string',
+    value : '/nix/store',
+    description : 'path of the Nix store (defaults to /nix/store)')
+
+
+# compiler args
+#============================================================================
+
+option(
+    'ldflags',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-L/usr/lib',
+        '-L/lib'],
+    description : 'Link flags')
+
+
+option(
+    'cxxflags',
+    type : 'array',
+    value : [
+        '-Wno-pedantic',
+        '-Wno-non-virtual-dtor',
+        '-Wno-unused-parameter',
+        '-Wno-variadic-macros',
+        '-Wdeprecated-declarations',
+        '-Wno-missing-field-initializers',
+    ],
+    description : 'C build flags')
+
+
+# optional dependencies
+#============================================================================
+
+option(
+    'enable_gc',
+    type : 'feature',
+    value : 'auto',
+    description : 'Build nix with Boehm garbage collector')
+
+option(
+    'enable_s3',
```

Comment:
Typically Meson options don't get prefixed with `enable` or `with`.

`-Denable_s3=enabled` vs `Ds3=enabled`.

Reply:
true, though again its just for parity with existing stuff :) can be changed

Reply:
I would focus on being ideomatic vs being the same. This PR will be some Nix dev's first introduction to Meson so it is good to demonstrate best practices.

Reply:
Indeed, the example table I wrote use `-Dgc=enabled` instead of `-Denable_gc=true`.